ticdc: improve description about ticdc compatibility with lightning & BR (#21609)
pingcap/docs: ticdc/ticdc-faq.md

## What are the compatibility limitations between TiDB Lightning Physical Import Mode and TiCDC?

TiDB Lightning [Physical Import Mode](/tidb-lightning/tidb-lightning-physical-import-mode.md) directly generates SST files and imports them into the TiKV cluster. Because this import mode does not include the regular data writing process, it does not generate change logs. In most cases, a changefeed cannot detect this kind of data change. Only during the initialization phase of the changefeed or when Region changes (such as split, merge, and leader transfer) trigger incremental scans might this kind of data change be detected. Therefore, a changefeed cannot fully capture the data imported via TiDB Lightning Physical Import Mode.

> If the tables operated by TiDB Lightning Physical Import Mode overlap with the tables monitored by the changefeed, unknown errors might occur due to incomplete data capture, such as a changefeed stalling and data inconsistency between upstream and downstream. If you need to use TiDB Lightning Physical Import Mode to import tables replicated by TiCDC, follow these steps:

Reviewer: ```suggestion
If the tables imported using TiDB Lightning Physical Import Mode overlap with the tables monitored by a changefeed, incomplete data capture might cause errors, such as stuck replication and data inconsistency between upstream and downstream. If you need to use TiDB Lightning Physical Import Mode to import tables replicated by TiCDC, follow these steps:
```

> 1. Remove the TiCDC replication task related to these tables.

Reviewer: ```suggestion
1. Delete the TiCDC replication task related to these tables.
```

> 2. Use TiDB Lightning Physical Import Mode to restore data separately in the upstream and downstream clusters of TiCDC.

Reviewer: ```suggestion
2. Use TiDB Lightning Physical Import Mode to import data into the upstream and downstream clusters of TiCDC respectively.
```

> 3. After the restoration is complete and the data consistency of the corresponding tables in the upstream and downstream clusters has been verified, use the timestamp (TSO) after the completion of TiDB Lightning Physical Import Mode as the `start-ts` of the TiCDC replication task to create a new TiCDC replication task for incremental replication.

Reviewer: ```suggestion
3. After the import is complete, verify the data consistency of the corresponding tables in the upstream and downstream clusters.

4. Create a new TiCDC replication task to resume incremental replication, using the timestamp (TSO) after the completion of the import as the `start-ts`.
```

    ```shell

>     cdc cli changefeed create -c "upstream-to-downstream-some-tables" --start-ts=431434047157698561 --sink-uri="mysql://root@127.0.0.1:4000? time-zone="

Reviewer: ```suggestion
    cdc cli changefeed create -c "upstream-to-downstream-some-tables" --start-ts=431434047157698561 --sink-uri="mysql://root@127.0.0.1:4000?time-zone="
```

    ```

> If you can confirm that the tables operated by TiDB Lightning Physical Import Mode do not overlap with the tables monitored by the changefeed, you can set the `check-requirements` in the TiDB Lightning configuration file to `false` to forcibly execute the data import operation.

Reviewer 2: ```suggestion
If you are sure that the tables operated by TiDB Lightning Physical Import Mode do not overlap with the tables monitored by the changefeed, you can set the `[check-requirements`](/tidb-lightning/tidb-lightning-configuration.md#check-requirements) in the TiDB Lightning configuration file to `false` to forcibly execute the data import operation.
```

## What are the compatibility limitations between BR (Backup & Restore) and TiCDC?

> BR (Backup & Restore) also directly generates SST files and imports them into the TiKV cluster. A changefeed cannot fully capture this kind of data change. For details, refer to [What are the compatibility limitations between TiDB Lightning Physical Import Mode and TiCDC?](/ticdc/ticdc-faq.md#what-are-the-compatibility-limitations-between-tidb-lightning-physical-import-mode-and-ticdc).

Author: ```suggestion
BR (Backup & Restore) also directly generates SST files and imports them into the TiKV cluster. A changefeed cannot fully capture data imported in this manner. For details, refer to [What are the compatibility limitations between TiDB Lightning Physical Import Mode and TiCDC?](/ticdc/ticdc-faq.md#what-are-the-compatibility-limitations-between-tidb-lightning-physical-import-mode-and-ticdc).
```
